core/src/Streamly/Internal/Data/Unboxed.hs

```diff
@@ -74,11 +75,22 @@ touch (ArrayContents contents) =
 --------------------------------------------------------------------------------
 
 #ifndef USE_STORABLE
+
+-- XXX This is a superset of Storable. Put a Storable constraint on this
+-- and remove sizeOf, alignment, readOffPtr, writeOffPtr from this. Just keep
+-- the bytearray reading/writing functions.
+
+-- | Keep the instances compatible with Storable.
 class Unboxed a where
     sizeOf :: a -> Int
     alignment :: a -> Int
+
+    -- Read an element of type "a" from a MutableByteArray given the element
+    -- index.
     readByteArray :: ArrayContents a -> Int -> IO a
     writeByteArray :: ArrayContents a -> Int -> a -> IO ()
+
+    -- Read an element of type "a" from a Ptr given the element index.
```

Comment:
Now that we use elements, it's pretty weird to use `Ptr` to consider element index.